# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [5]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [6]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3         V4        V5        V6        V7       
1 0.8369779 0.09212071 0.09214004 0.2254914 0.7904512 0.7401084 0.8751569
2 0.4031544 0.05707795 0.07855167 0.4570784 0.0201527 0.6520688 0.3404492
  V8         V9        V10       V11       V12        V13       V14       
1 0.05332528 0.6004130 0.5348147 0.4787272 0.75108317 0.3153938 0.08845412
2 0.05918284 0.8310017 0.6772946 0.4207675 0.08134792 0.5261547 0.50356146
  V15       V16       V17       V18       V19       V20      
1 0.3973590 0.6566461 0.7729104 0.8611769 0.8455865 0.2843350
2 0.6712287 0.1741355 0.7326056 0.2941587 0.1333820 0.6100256

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [7]:
df2 <- df1 %>%
  gather(Key, Value, V1 : V20)

head(df2)

Key Value     
1 V1  0.83697790
2 V1  0.40315444
3 V2  0.09212071
4 V2  0.05707795
5 V3  0.09214004
6 V3  0.07855167

## Assignment 2

We will generate some data:

In [8]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [9]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [14]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook